File: Put Marbles in Bags/Put Marbles in Bags.py
# Solution for Put Marbles in Bags in PY

class Solution:
    def putMarbles(self, weights: List[int], k: int) -> int:
        n = len(weights)
        pairSum = [0] * (n - 1)
        for i in range(n - 1):
            pairSum[i] = weights[i] + weights[i + 1]
        pairSum.sort()
        minSum, maxSum = 0, 0
        for i in range(k - 1):
            minSum += pairSum[i]
            maxSum += pairSum[n - 1 - i - 1]
        return maxSum - minSum

File: Put Marbles in Bags/Put Marbles in Bags.java
// Solution for Put Marbles in Bags in JAVA

class Solution {
    public long putMarbles(int[] weights, int k) {
        int n = weights.length;
        int[] pairSum = new int[n - 1];
        for (int i = 0; i < n - 1; i++) {
            pairSum[i] = weights[i] + weights[i + 1];
        }
        Arrays.sort(pairSum);
        long minSum = 0, maxSum = 0;
        for (int i = 0; i < k - 1; i++) {
            minSum += pairSum[i];
            maxSum += pairSum[n - 1 - i - 1];
        }
        return maxSum - minSum;
    }
}
